content/blog/2024/11/04/2024-11-04-jenkins-tutorial-revamp.adoc

```diff
@@ -0,0 +1,273 @@
+
+This command, while functional, is overwhelming for beginners and lacks context.
+Why do we need a Docker network?
+Why are we running Docker in Docker?
+These questions were left unanswered.
+
+== The Solution: Revamping with Docker Compose
+
+After much deliberation and feedback, we decided to leverage Docker Compose to simplify the entire setup process. 
+
+=== Why Docker Compose?
+
+Docker Compose can encapsulate complex multi-container setups into a single, easy-to-understand file.
+It was the only tool that could handle the complexity of the Jenkins controller and agent setup while making it accessible to newcomers.
+
+Here's an example of how Docker Compose simplifies things:
+
+[source,yaml]
+----
+services:
+  jenkins:
+    image: jenkins/jenkins:lts
+    ports:
+      - "8080:8080"
+    volumes:
+      - jenkins_home:/var/jenkins_home
+  agent:
+    image: jenkins/ssh-agent:jdk17
+    environment:
+      - JENKINS_AGENT_SSH_PUBKEY=<your_public_key>
+volumes:
+  jenkins_home:
+----
+
+With this setup, users can start Jenkins and an agent with a single command: `docker compose up -d`.
+
+=== Key Improvements
+
+1. *Simplified setup*: Instead of a series of complex Docker commands, users can now get started with a single Docker Compose command.
+
+2. *Reduced assumptions*: By encapsulating the setup in Docker Compose, we removed the need for extensive Docker knowledge.
+
+3. *Focus on Jenkins*: With a streamlined setup, users can dive into learning Jenkins faster.
+
+4. *Modernized tools*: We replaced outdated plugins like BlueOcean with more current solutions.
+
+5. *Elimination of Docker in Docker*: Our new setup avoids this security risk entirely.
+
+6. *Contextual guidance*: We've added explanations for each step, helping users understand why they're performing certain actions.
+
+== The Process: Collaboration and Innovation
+
+This revamp was a collaborative effort, with significant contributions from our Google Summer of Code participant, author:ash-sxn[Ashutosh Saxena].
+His fresh perspective as a newcomer to Jenkins was invaluable in identifying pain points that we, as experienced developers, had overlooked.
+
+=== Key Innovations
+
+1. *Multi-architecture support*: We ensured our Docker images work on both `x86_64` and `ARM64` architectures.
+This was a significant challenge that Ashutosh tackled using Docker's `buildx` tool.
+
+2. *GitPod integration*: We made Jenkins accessible directly from the browser, reducing hardware barriers to entry.
+This was particularly important for users with less powerful machines or those in environments with limited resources.
+
+3. *Docker Compose profiles*: We introduced profiles for different languages and technologies, making it easy to switch between setups.
+For example, users can start a Maven-specific setup with `docker compose --profile maven up -d`.
+
+4. *Automated updates*: We implemented automation to keep our Docker images up-to-date, using tools like Dependabot and UpdateCLI.
```

Comment:
When I'm curious about a first mention of a technology, I like it when there is a hyperlink that tells me more.
```suggestion
4. *Automated updates*: We implemented automation to keep our Docker images up-to-date, using tools like link:https://docs.github.com/en/code-security/dependabot[Dependabot] and link:https://www.updatecli.io/[UpdateCLI].
```